File: app/src/main/java/intellij/class5/objetos/exerciseadditional/java/Estadio.java
package intellij.class5.objetos.exerciseadditional.java;

import java.util.ArrayList;
import java.util.List;

public class Estadio {
    private List<Prova> provaList;

    private Prova provaHabilidade = new Prova();
    private Prova provaCorrida = new Prova();
    private Prova provaLideranca = new Prova();
    private Prova provaPuloAlto = new Prova();
    private Prova provaLanceLivre = new Prova();
    private Prova provaJogoEmEquipe = new Prova();

    public Estadio() {
    }

    public Estadio(List<Prova> provaList) {
        this.provaList = provaList;
    }

    public List<Boolean> saberDesempenho(Atleta atleta) {
        List<Boolean> provasRealizadas = new ArrayList<>();
        for (int index = 0; index < getProvaList().size(); index++) {
            provasRealizadas.add(getProvaList().get(index).podeRealizar(atleta));
        }
        return provasRealizadas;
    }

    public String whoWinsTheGoldMedal(Estadio this, Atleta atleta1, Atleta atleta2) {
        if (this.saberDesempenho(atleta1).size() == this.saberDesempenho(atleta2).size()) {
            if (atleta1.getNivel() > atleta2.getNivel()) return atleta1.getName();
            else return atleta2.getName();
        } else if (this.saberDesempenho(atleta1).size() > this.saberDesempenho(atleta2).size()) {
            return atleta1.getName();
        } else
            return atleta2.getName();
    }

    private List<Prova> createProvasSerie1() {
        provaHabilidade.setDificuldade(3);
        provaHabilidade.setEnergiaNecessaria(75);
        provaCorrida.setDificuldade(5);
        provaCorrida.setEnergiaNecessaria(135);
        provaLideranca.setDificuldade(10);
        provaLideranca.setEnergiaNecessaria(200);
        provaPuloAlto.setDificuldade(4);
        provaPuloAlto.setEnergiaNecessaria(95);

        List<Prova> provaList = new ArrayList<>();
        provaList.add(provaHabilidade);
        provaList.add(provaCorrida);
        provaList.add(provaLideranca);
        provaList.add(provaPuloAlto);
        return provaList;
    }

    private List<Prova> createProvasSerie2() {
        provaCorrida.setDificuldade(9);
        provaCorrida.setEnergiaNecessaria(200);
        provaPuloAlto.setDificuldade(6);
        provaPuloAlto.setEnergiaNecessaria(145);
        provaLanceLivre.setDificuldade(8);
        provaLanceLivre.setEnergiaNecessaria(156);
        provaJogoEmEquipe.setDificuldade(3);
        provaJogoEmEquipe.setEnergiaNecessaria(75);

        List<Prova> provaList = new ArrayList<>();
        provaList.add(provaCorrida);
        provaList.add(provaPuloAlto);
        provaList.add(provaLanceLivre);
        provaList.add(provaJogoEmEquipe);
        return provaList;
    }

    public List<Prova> getProvaList() {
        return provaList;
    }

    public void setProvaList(int number) {
        switch (number) {
            case 0:
                this.provaList = createProvasSerie1();
            case 1:
                this.provaList = createProvasSerie2();
        }
    }
}

File: app/src/main/java/intellij/class5/objetos/exerciseadditional/kotlin/Estadio.kt
package intellij.class5.objetos.exerciseadditional.kotlin

import java.util.ArrayList

class Estadio {
    var provaList: List<Prova> = listOf()
    private val provaHabilidade = Prova()
    private val provaCorrida = Prova()
    private val provaLideranca = Prova()
    private val provaPuloAlto = Prova()
    private val provaLanceLivre = Prova()
    private val provaJogoEmEquipe = Prova()

    constructor()
    constructor(provaList: List<Prova>) {
        this.provaList = provaList
    }

    fun saberDesempenho(atleta: Atleta): List<Boolean> {
//        List<Prova> provasRealizadas = new ArrayList<>();
        val provasRealizadas: MutableList<Boolean> = ArrayList()
        for (index in provaList.indices) {
//            if (getProvaList().get(index).podeRealizar(atleta)) {
//                provasRealizadas.add(getProvaList().get(index));
//            }
            provasRealizadas.add(provaList[index].podeRealizar(atleta))
        }
        return provasRealizadas
    }

    fun whoWinsTheGoldMedal(atleta1: Atleta, atleta2: Atleta): String {
        return if (saberDesempenho(atleta1).size == saberDesempenho(atleta2).size) {
            if (atleta1.nivel > atleta2.nivel) atleta1.name else atleta2.name
        } else if (saberDesempenho(atleta1).size > saberDesempenho(atleta2).size) {
            atleta1.name
        } else atleta2.name
    }

    private fun createProvasSerie1(): List<Prova> {
        provaHabilidade.apply {
            dificuldade = 3
            energiaNecessaria = 75
        }
        provaCorrida.apply {
            dificuldade = 5
            energiaNecessaria = 135
        }
        provaLideranca.apply {
            dificuldade = 10
            energiaNecessaria = 200
        }
        provaPuloAlto.apply {
            dificuldade = 4
            energiaNecessaria = 95
        }
        val provaList: MutableList<Prova> = mutableListOf()
        provaList.add(provaHabilidade)
        provaList.add(provaCorrida)
        provaList.add(provaLideranca)
        provaList.add(provaPuloAlto)
        return provaList
    }

    private fun createProvasSerie2(): List<Prova> {
        provaCorrida.apply {
            dificuldade = 9
            energiaNecessaria = 200
        }
        provaPuloAlto.apply {
            dificuldade = 6
            energiaNecessaria = 145
        }
        provaLanceLivre.apply {
            dificuldade = 8
            energiaNecessaria = 156
        }
        provaJogoEmEquipe.apply {
            dificuldade = 3
            energiaNecessaria = 75
        }
        val provaList: MutableList<Prova> = ArrayList()
        provaList.add(provaCorrida)
        provaList.add(provaPuloAlto)
        provaList.add(provaLanceLivre)
        provaList.add(provaJogoEmEquipe)
        return provaList
    }

    fun setProvaList(number: Int) {
        when (number) {
            0 -> provaList = createProvasSerie1()
            1 -> provaList = createProvasSerie2()
        }
    }
}